components/memory-queue/src/types.ts

```diff
@@ -0,0 +1,28 @@
+export interface ReceiveMessagesOptions {
+  visibilityTimeout?: number
+  waitTimeSeconds?: number
+  abortSignal?: AbortSignal
+}
+
+export interface QueueMessage {
+  MessageId: string
+  ReceiptHandle: string
+  Body: string
+}
+
+export interface QueueStatus {
+  ApproximateNumberOfMessages: string
+  ApproximateNumberOfMessagesNotVisible: string
+  ApproximateNumberOfMessagesDelayed: string
+}
+
+export interface IQueueComponent {
```

Comment:
Sharing the same interface between the sqs component should make it easier to interchange. WDYT about doing what was done to the cache component and put the interface in the commons dependency?